lite/core/mir/pattern_matcher.h

```diff
@@ -157,8 +157,16 @@ struct PMNode {
 
   template <typename T>
   PMNode* assert_op_attr(const std::string& attr_name, const T& attr) {
-    return assert_op_attr_satisfied<T>(
-        attr_name, [&](const T& src) { return src == attr; });
+    asserts_.push_back([=](const Node* x) {
+      if (x && x->IsStmt()) {
+        auto* op_info = x->stmt()->op_info();
+        bool cond = (op_info->HasAttr(attr_name) &&
+                     op_info->GetAttr<T>(attr_name) == attr);
+        return cond;
+      }
+      return false;
+    });
+    return this;
```

Comment:
why change this ？

Reply:
There are some passes under CUDA that have a segment-fault in this function.

Reply:
thanks, we have found the problem, we are fixing.

Reply:
OK!